File: 2028 find-missing-observations/solution.cpp
class Solution {
public:
    vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
        int m = rolls.size();
        int sum = 0;
        for(int i:rolls) sum+=i;
        int an = mean*(m+n)-sum;
        if(an < n || an > n*6) return {};
        vector<int> ans(n);
        int b=an/n,r=an%n;
        for(int i=0;i<n;i++) {
            ans[i]=b+( i<r ? 1 : 0);
            if (ans[i] < 1 || ans[i] > 6) return {};
        }
        return ans;
    }
};

File: 2028 find-missing-observations/solution.rs
impl Solution {
    pub fn missing_rolls(rolls: Vec<i32>, mean: i32, n: i32) -> Vec<i32> {
        let m = rolls.len() as i32;
        let sum_observed: i32 = rolls.iter().sum();
        let total = mean * (m + n);
        let missing_sum = total - sum_observed;
        if missing_sum < n || missing_sum > 6 * n {
            return vec![]; 
        }
        let base = missing_sum / n;
        let rem = missing_sum % n;
        let mut ans = vec![base; n as usize];
        for i in 0..rem {
            ans[i as usize] += 1;
        }
        if ans.iter().any(|&x| x < 1 || x > 6) {
            return vec![];
        }
        ans
    }
}

